src/mesh/MFEMMesh.C

```diff
@@ -16,14 +17,143 @@ defineLegacyParams(MFEMMesh);
 InputParameters
 MFEMMesh::validParams()
 {
-  InputParameters params = GeneratedMesh::validParams();
+  InputParameters params = MooseMesh::validParams();
   params.addRequiredParam<MeshFileName>("file", "The name of the mesh file to read");
+  params.addParam<bool>("CreateMOOSEMesh", true, "Allows the user to choose whether to create the corresponding MOOSE mesh, or create a dummy mesh instead");
```

Comment:
Would it be easier to split into two files - `MFEMMesh` and `CoupledMFEMMesh` - and then rather than have this bool, the user would just select which of the two objects types to use. `CoupledMFEMMesh` could inherit from `MFEMMesh,` but with an overridden `buildMesh` method.

This would also get rid of the need for `buildDummyMesh` and `buildRealMesh`... I don't like `buildRealMesh` as a name, as I think it's confusing when we'll be talking about real/complex variables in similar contexts.